src/Signature/SignatureV4.php

```diff
@@ -98,7 +154,7 @@ public function presign(
         $parsed['query']['X-Amz-Algorithm'] = 'AWS4-HMAC-SHA256';
         $parsed['query']['X-Amz-Credential'] = $credential;
         $parsed['query']['X-Amz-Date'] = gmdate('Ymd\THis\Z', $startTimestamp);
-        $parsed['query']['X-Amz-SignedHeaders'] = 'host';
+        $parsed['query']['X-Amz-SignedHeaders'] = join(';', $this->getPresignHeaders($parsed['headers']));
```

Comment:
Can you use `implode` here over its alias function `join`? This line being over 80 is okay since changing it would make this section harder to read.